### PR TITLE
Replace part of network functions on platform-independent implementation

### PR DESCRIPTION
**Pull request by Diadlo**
This change is [Reviewable](https://reviewable.io/reviews/toktok/c-toxcore/461)

### CODE REVIEW

**Comment by sudden6** [comment]
There are many tabs where spaces should be.

---







*Comments from [Reviewable](https://reviewable.io:443/reviews/toktok/c-toxcore/461#-:-Kb_pFoKNcMRUktPMxZt:b4uhlgq)*

**Comment by Diadlo** [comment]
@sudden6 Sorry. Can you check now?

**Comment by sudden6** [comment]
There are still tabs in:
crypto_test.c
network_test.c
bootstrap_node_packets.c
DHT_bootstrap.c

---







*Comments from [Reviewable](https://reviewable.io:443/reviews/toktok/c-toxcore/461#-:-Kb_tc48GSvx7cp5btJX:b-bw16rr)*

**Comment by Diadlo** [comment]
@sudden6 Thanks a lot! Fixed.

**Comment by sudden6** [comment]
I think you forgot to upload the changes^^

**Comment by Diadlo** [comment]
@sudden6 Hm.. Seems updated

**Comment by sudden6** [comment]
ok, I see it now too, maybe missed it before

**Comment by sudden6** [comment]
Reviewed 2 of 29 files at r1, 2 of 7 files at r2, 2 of 13 files at r3.
Review status: 6 of 29 files reviewed at latest revision, 5 unresolved discussions, some commit checks failed.

---

*[toxcore/network.c, line 1239 at r3](https://reviewable.io:443/reviews/toktok/c-toxcore/461#-KbbBktaTWrZtWLwVf0I:-KbbBktaTWrZtWLwVf0J:by7jpnl) ([raw file](https://github.com/toktok/c-toxcore/blob/cb13fb57136eb9b33457d512369bdd292cab21aa/toxcore/network.c#L1239)):*
> ```C
>     }
> 
>     res = malloc(sizeof(IP_Port) * count);
> ```

check return value?

---

*[toxcore/network.c, line 1348 at r3](https://reviewable.io:443/reviews/toktok/c-toxcore/461#-KbbB3MDyLev7sMh11me:-KbbB3MEuwkuaYfSQ2D9:bp8noln) ([raw file](https://github.com/toktok/c-toxcore/blob/cb13fb57136eb9b33457d512369bdd292cab21aa/toxcore/network.c#L1348)):*
> ```C
>     target.sin_family = make_family(ip_port.ip.family);
>     target.sin_port = net_htons(ip_port.port);
>     fill_addr4(ip_port.ip.ip4, &target.sin_addr);
> ```

only IPv4?

---

*[toxcore/network.h, line 388 at r3](https://reviewable.io:443/reviews/toktok/c-toxcore/461#-KbbD-rMmDRMrjVAKLQZ:-KbbD-rMmDRMrjVAKLQ_:b-hjtm5y) ([raw file](https://github.com/toktok/c-toxcore/blob/cb13fb57136eb9b33457d512369bdd292cab21aa/toxcore/network.h#L388)):*
> ```objc
> 
> /* High-level getaddrinfo implementation.
>  * Given node, which identify an Internet host. net_getipport() fill array with
> ```

`which identifies an Internet host.` I don't really understand the rest of the paragraph, can you rephrase/explain it?

---

*[toxcore/network.h, line 392 at r3](https://reviewable.io:443/reviews/toktok/c-toxcore/461#-KbbDKHMwJ8C6HAqYmxZ:-KbbDKHMwJ8C6HAqYmx_:b-swaf50) ([raw file](https://github.com/toktok/c-toxcore/blob/cb13fb57136eb9b33457d512369bdd292cab21aa/toxcore/network.h#L392)):*
> ```objc
>  * that can be specified in a call to net_connect().
>  *
>  * Skip all addresses with socktype != type (use type = -1 to grep all address)
> ```

`to grep all addresses` ?

---

*[toxcore/network.h, line 393 at r3](https://reviewable.io:443/reviews/toktok/c-toxcore/461#-KbbDiG7KyzreGLus2Hm:-KbbDiG8-XzeyMfLXYT8:bo41vao) ([raw file](https://github.com/toktok/c-toxcore/blob/cb13fb57136eb9b33457d512369bdd292cab21aa/toxcore/network.h#L393)):*
> ```objc
>  *
>  * Skip all addresses with socktype != type (use type = -1 to grep all address)
>  * To correct deallocate array memory use net_freeipport()
> ```

`To correctly deallocate`?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/toktok/c-toxcore/461)*

**Comment by sudden6** [comment]
Reviewed 9 of 29 files at r1, 2 of 7 files at r2, 10 of 13 files at r3.
Review status: 27 of 29 files reviewed at latest revision, 5 unresolved discussions, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/toktok/c-toxcore/461)*

**Comment by Diadlo** [comment]
Review status: 25 of 30 files reviewed at latest revision, 5 unresolved discussions, some commit checks failed.

---

*[toxcore/network.c, line 1239 at r3](https://reviewable.io:443/reviews/toktok/c-toxcore/461#-KbbBktaTWrZtWLwVf0I:-KbfnIIgJ944pk7HvBBq:b-896fix) ([raw file](https://github.com/toktok/c-toxcore/blob/30a7fbb692bb4bc80b936939d5776b79706226f7/toxcore/network.c#L1239)):*
<details><summary><i>Previously, sudden6 wrote…</i></summary><blockquote>

check return value?
</blockquote></details>

Done.

---

*[toxcore/network.c, line 1348 at r3](https://reviewable.io:443/reviews/toktok/c-toxcore/461#-KbbB3MDyLev7sMh11me:-KbfpRPp0sk6I7_yKEkk:brun27l) ([raw file](https://github.com/toktok/c-toxcore/blob/30a7fbb692bb4bc80b936939d5776b79706226f7/toxcore/network.c#L1348)):*
<details><summary><i>Previously, sudden6 wrote…</i></summary><blockquote>

only IPv4?
</blockquote></details>

Yes. Function renamed and TODO added.

---

*[toxcore/network.h, line 388 at r3](https://reviewable.io:443/reviews/toktok/c-toxcore/461#-KbbD-rMmDRMrjVAKLQZ:-KbftkC_muNIuaTu5Sx4:br0bztg) ([raw file](https://github.com/toktok/c-toxcore/blob/30a7fbb692bb4bc80b936939d5776b79706226f7/toxcore/network.h#L388)):*
<details><summary><i>Previously, sudden6 wrote…</i></summary><blockquote>

`which identifies an Internet host.` I don't really understand the rest of the paragraph, can you rephrase/explain it?
</blockquote></details>

Better now?

---

*[toxcore/network.h, line 392 at r3](https://reviewable.io:443/reviews/toktok/c-toxcore/461#-KbbDKHMwJ8C6HAqYmxZ:-KbfuF5wigBU93jW95Uc:b-896fix) ([raw file](https://github.com/toktok/c-toxcore/blob/30a7fbb692bb4bc80b936939d5776b79706226f7/toxcore/network.h#L392)):*
<details><summary><i>Previously, sudden6 wrote…</i></summary><blockquote>

`to grep all addresses` ?
</blockquote></details>

Done.

---

*[toxcore/network.h, line 393 at r3](https://reviewable.io:443/reviews/toktok/c-toxcore/461#-KbbDiG7KyzreGLus2Hm:-KbfuEfI4zaU3AtngayE:b-896fix) ([raw file](https://github.com/toktok/c-toxcore/blob/30a7fbb692bb4bc80b936939d5776b79706226f7/toxcore/network.h#L393)):*
<details><summary><i>Previously, sudden6 wrote…</i></summary><blockquote>

`To correctly deallocate`?
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/toktok/c-toxcore/461)*

**Comment by sudden6** [comment]
Reviewed 1 of 7 files at r4.
Review status: 24 of 30 files reviewed at latest revision, 5 unresolved discussions, some commit checks failed.

---

*[auto_tests/network_test.c, line 58 at r4](https://reviewable.io:443/reviews/toktok/c-toxcore/461#-Kc47x1gl0B9z3dNFwSi:-Kc47x1gl0B9z3dNFwSj:bdftn2f) ([raw file](https://github.com/toktok/c-toxcore/blob/a477e838a5a31c67d250460475a94bb31b57b596/auto_tests/network_test.c#L58)):*
> ```C
> 
>     if (localhost_split) {
>         printf("Localhost seems to be split in two.\n");
> ```

This error message doesn't make sense to me, what does `split in two` mean?

---

*[toxcore/crypto_core.c, line 217 at r4](https://reviewable.io:443/reviews/toktok/c-toxcore/461#-Kc44xGR_h0FxkKvhxfQ:-Kc44xGR_h0FxkKvhxfR:btxph1n) ([raw file](https://github.com/toktok/c-toxcore/blob/a477e838a5a31c67d250460475a94bb31b57b596/toxcore/crypto_core.c#L217)):*
> ```C
> #if BYTE_ORDER == LITTLE_ENDIAN
>     uint8_t *s = (uint8_t *)&x;
>     return (uint32_t)(s[0] << 24 | s[1] << 16 | s[2] << 8 | s[3]);
> ```

Is there a reason why you used the way with casts instead of the "shift and or" way to change the byte order? Also why does crypto_core need it's own implementation of byte order swapping? If there's a good reason for it, can you add a comment?

---

*[toxcore/network.c, line 1236 at r4](https://reviewable.io:443/reviews/toktok/c-toxcore/461#-Kc45PNokaAfQBGlAJY0:-Kc45PNqxWYwcECIhN9y:ber1vxl) ([raw file](https://github.com/toktok/c-toxcore/blob/a477e838a5a31c67d250460475a94bb31b57b596/toxcore/network.c#L1236)):*
> ```C
>     *res = (IP_Port*)malloc(sizeof(IP_Port) * count);
>     if (*res == NULL) {
>         return -1;
> ```

returning `-1` from an `unsigned int` function is wrong

---

*[toxcore/network.h, line 388 at r3](https://reviewable.io:443/reviews/toktok/c-toxcore/461#-KbbD-rMmDRMrjVAKLQZ:-Kc46y95hG5Vjxsq9p0i:bt5p0r7) ([raw file](https://github.com/toktok/c-toxcore/blob/30a7fbb692bb4bc80b936939d5776b79706226f7/toxcore/network.h#L388)):*
<details><summary><i>Previously, Diadlo (Polshakov Dmitry) wrote…</i></summary><blockquote>

Better now?
</blockquote></details>

Better but not quite there yet. Which format does `node` have?

---

*[toxcore/network.h, line 392 at r4](https://reviewable.io:443/reviews/toktok/c-toxcore/461#-Kc47TG3XG5aT981O0cv:-Kc47TG4sP4fbdtdcUxn:b-gct5fc) ([raw file](https://github.com/toktok/c-toxcore/blob/a477e838a5a31c67d250460475a94bb31b57b596/toxcore/network.h#L392)):*
> ```objc
>  * To correctly deallocate array memory use net_freeipport()
>  *
>  * return size of res array.
> ```

size in bytes or number of entries?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/toktok/c-toxcore/461)*

**Comment by Diadlo** [comment]
Review status: 24 of 30 files reviewed at latest revision, 5 unresolved discussions, some commit checks failed.

---

*[auto_tests/network_test.c, line 58 at r4](https://reviewable.io:443/reviews/toktok/c-toxcore/461#-Kc47x1gl0B9z3dNFwSi:-Kc74Z3rPdHI9EBTVVCa:btxzu9d) ([raw file](https://github.com/toktok/c-toxcore/blob/a477e838a5a31c67d250460475a94bb31b57b596/auto_tests/network_test.c#L58)):*
<details><summary><i>Previously, sudden6 wrote…</i></summary><blockquote>

This error message doesn't make sense to me, what does `split in two` mean?
</blockquote></details>

For me too. But it's out of the scope of this PR.

---

*[toxcore/crypto_core.c, line 217 at r4](https://reviewable.io:443/reviews/toktok/c-toxcore/461#-Kc44xGR_h0FxkKvhxfQ:-Kc74q5jXWfjIj6-an9n:bpa177n) ([raw file](https://github.com/toktok/c-toxcore/blob/a477e838a5a31c67d250460475a94bb31b57b596/toxcore/crypto_core.c#L217)):*
> Is there a reason why you used the way with casts instead of the "shift and or" way to change the byte order?

I use cast and "shift and or". Or I don't understand you.

> Also why does crypto_core need it's own implementation of byte order swapping?

Because `htonl` is implemented in `#include <arpa/inet.h>` we don't want to depend of it

---

*[toxcore/network.c, line 1236 at r4](https://reviewable.io:443/reviews/toktok/c-toxcore/461#-Kc45PNokaAfQBGlAJY0:-Kc75j8yNKJvF-NV3bcB:b-9zdwtd) ([raw file](https://github.com/toktok/c-toxcore/blob/a477e838a5a31c67d250460475a94bb31b57b596/toxcore/network.c#L1236)):*
<details><summary><i>Previously, sudden6 wrote…</i></summary><blockquote>

returning `-1` from an `unsigned int` function is wrong
</blockquote></details>

-1 is invalid value and indicate invalid function result. It still can be checked with`res != -1`
I don't want to lose half of integer just because we need indicator for invalid result

---

*[toxcore/network.h, line 388 at r3](https://reviewable.io:443/reviews/toktok/c-toxcore/461#-KbbD-rMmDRMrjVAKLQZ:-Kc76CWIMJLf74RTtQ4I:b-do3zdc) ([raw file](https://github.com/toktok/c-toxcore/blob/30a7fbb692bb4bc80b936939d5776b79706226f7/toxcore/network.h#L388)):*
<details><summary><i>Previously, sudden6 wrote…</i></summary><blockquote>

Better but not quite there yet. Which format does `node` have?
</blockquote></details>

It's part of `getaddrinfo` [docs](https://linux.die.net/man/3/getaddrinfo). So it will be retranslated to `getaddrinfo` with their node meaning

---

*[toxcore/network.h, line 392 at r4](https://reviewable.io:443/reviews/toktok/c-toxcore/461#-Kc47TG3XG5aT981O0cv:-Kc76ktoblaMwQCUFMBN:b-lsbneg) ([raw file](https://github.com/toktok/c-toxcore/blob/a477e838a5a31c67d250460475a94bb31b57b596/toxcore/network.h#L392)):*
<details><summary><i>Previously, sudden6 wrote…</i></summary><blockquote>

size in bytes or number of entries?
</blockquote></details>

AFAIK 'size of array' is alwais mean number of elements (exclude `sizeof` in C)

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/toktok/c-toxcore/461)*

**Comment by sudden6** [comment]
Reviewed 3 of 7 files at r4.
Review status: 27 of 30 files reviewed at latest revision, 4 unresolved discussions, some commit checks failed.

---

*[toxcore/crypto_core.c, line 217 at r4](https://reviewable.io:443/reviews/toktok/c-toxcore/461#-Kc44xGR_h0FxkKvhxfQ:-Kc8ZPTCnUTCBiVWzuE9:b-5qkg6a) ([raw file](https://github.com/toktok/c-toxcore/blob/a477e838a5a31c67d250460475a94bb31b57b596/toxcore/crypto_core.c#L217)):*
<details><summary><i>Previously, Diadlo (Polshakov Dmitry) wrote…</i></summary><blockquote>

> Is there a reason why you used the way with casts instead of the "shift and or" way to change the byte order?

I use cast and "shift and or". Or I don't understand you.

> Also why does crypto_core need it's own implementation of byte order swapping?

Because `htonl` is implemented in `#include <arpa/inet.h>` we don't want to depend of it
</blockquote></details>

I would have done it like this
```swapped = ((num>>24)&0x000000FF) | // move byte 3 to byte 0
                    ((num<<8)&0x00FF0000) | // move byte 1 to byte 2
                    ((num>>8)&0x0000FF00) | // move byte 2 to byte 1
                    ((num<<24)&0xFF000000); // byte 0 to byte 3 ```

this way there's no need to cast anything, which is better I think
```


---

*[toxcore/network.c, line 1236 at r4](https://reviewable.io:443/reviews/toktok/c-toxcore/461#-Kc45PNokaAfQBGlAJY0:-Kc8_XXpK3KeuMZ7lGQ8:b-7216av) ([raw file](https://github.com/toktok/c-toxcore/blob/a477e838a5a31c67d250460475a94bb31b57b596/toxcore/network.c#L1236)):*
<details><summary><i>Previously, Diadlo (Polshakov Dmitry) wrote…</i></summary><blockquote>

-1 is invalid value and indicate invalid function result. It still can be checked with`res != -1`
I don't want to lose half of integer just because we need indicator for invalid result
</blockquote></details>

Then you'd need to make the invalid value `UINT_MAX` or use `0` as error value, or use an error paramenter. I think the way you're doing it currently is not a good practice or defined in the C standard, but prove me wrong :)

---

*[toxcore/network.h, line 392 at r4](https://reviewable.io:443/reviews/toktok/c-toxcore/461#-Kc47TG3XG5aT981O0cv:-Kc8bhLW7K95CSYbbGIL:b-nvl9hh) ([raw file](https://github.com/toktok/c-toxcore/blob/a477e838a5a31c67d250460475a94bb31b57b596/toxcore/network.h#L392)):*
<details><summary><i>Previously, Diadlo (Polshakov Dmitry) wrote…</i></summary><blockquote>

AFAIK 'size of array' is alwais mean number of elements (exclude `sizeof` in C)
</blockquote></details>

I'd prefer `number of elements in res array` to avoid any ambiguity, but I'll leave it up to you.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/toktok/c-toxcore/461)*

**Comment by Diadlo** [comment]
Review status: 27 of 30 files reviewed at latest revision, 4 unresolved discussions.

---

*[toxcore/crypto_core.c, line 217 at r4](https://reviewable.io:443/reviews/toktok/c-toxcore/461#-Kc44xGR_h0FxkKvhxfQ:-KcADKBR-bY3Miz2A5Xe:bg3kesx) ([raw file](https://github.com/toktok/c-toxcore/blob/a477e838a5a31c67d250460475a94bb31b57b596/toxcore/crypto_core.c#L217)):*
<details><summary><i>Previously, sudden6 wrote…</i></summary><blockquote>

I would have done it like this
```swapped = ((num>>24)&0x000000FF) | // move byte 3 to byte 0
                    ((num<<8)&0x00FF0000) | // move byte 1 to byte 2
                    ((num>>8)&0x0000FF00) | // move byte 2 to byte 1
                    ((num<<24)&0xFF000000); // byte 0 to byte 3 ```

this way there's no need to cast anything, which is better I think
</blockquote></details>

I think, previous variant was more readable. But done.

---

*[toxcore/network.c, line 1236 at r4](https://reviewable.io:443/reviews/toktok/c-toxcore/461#-Kc45PNokaAfQBGlAJY0:-KcADQs_3LDE__cjDX-g:b-f2evy7) ([raw file](https://github.com/toktok/c-toxcore/blob/a477e838a5a31c67d250460475a94bb31b57b596/toxcore/network.c#L1236)):*
<details><summary><i>Previously, sudden6 wrote…</i></summary><blockquote>

Then you'd need to make the invalid value `UINT_MAX` or use `0` as error value, or use an error paramenter. I think the way you're doing it currently is not a good practice or defined in the C standard, but prove me wrong :)
</blockquote></details>

Changed return type. Done.

---

*[toxcore/network.h, line 392 at r4](https://reviewable.io:443/reviews/toktok/c-toxcore/461#-Kc47TG3XG5aT981O0cv:-KcADRLfAonEl7gq7e5Z:b-896fix) ([raw file](https://github.com/toktok/c-toxcore/blob/a477e838a5a31c67d250460475a94bb31b57b596/toxcore/network.h#L392)):*
<details><summary><i>Previously, sudden6 wrote…</i></summary><blockquote>

I'd prefer `number of elements in res array` to avoid any ambiguity, but I'll leave it up to you.
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/toktok/c-toxcore/461)*
<!-- Sent from Reviewable.io -->

**Comment by sudden6** [comment]
*[toxcore/crypto_core.c, line 217 at r4](https://reviewable.io:443/reviews/toktok/c-toxcore/461#-Kc44xGR_h0FxkKvhxfQ:-KcALDmQoz0ae2F9hSzd:b-9ux9mu) ([raw file](https://github.com/toktok/c-toxcore/blob/a477e838a5a31c67d250460475a94bb31b57b596/toxcore/crypto_core.c#L217)):*
<details><summary><i>Previously, Diadlo (Polshakov Dmitry) wrote…</i></summary><blockquote>

I think, previous variant was more readable. But done.
</blockquote></details>

sorry I forgot one line, `((num>>24)&0x000000FF) | // move byte 3 to byte 0` has to be the first one

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/toktok/c-toxcore/461)*

**Comment by sudden6** [comment]
Reviewed 3 of 4 files at r5.
Review status: 29 of 30 files reviewed at latest revision, 2 unresolved discussions, some commit checks failed.

---

*[toxcore/crypto_core.c, line 219 at r6](https://reviewable.io:443/reviews/toktok/c-toxcore/461#-KcCbZNpNTUfKodhRYYw:-KcCbZNqA8Hpw1wWozCq:b-t3hmex) ([raw file](https://github.com/toktok/c-toxcore/blob/2f8b53af89a6b25957ca647eb09446257d017ed9/toxcore/crypto_core.c#L219)):*
> ```C
>         ((x >> 24) & 0x000000FF) | // move byte 3 to byte 0
>         ((x << 8)  & 0x00FF0000) | // move byte 1 to byte 2
>         ((x >> 8)  & 0x0000FF00) | // move byte 2 to byte 1
> ```

maybe you could switch this line and the line above, so it's 3,2,1,0

---

*[toxcore/network.h, line 388 at r3](https://reviewable.io:443/reviews/toktok/c-toxcore/461#-KbbD-rMmDRMrjVAKLQZ:-KcCby-jQUN44tOgJiqA:b-3zjs5w) ([raw file](https://github.com/toktok/c-toxcore/blob/30a7fbb692bb4bc80b936939d5776b79706226f7/toxcore/network.h#L388)):*
<details><summary><i>Previously, Diadlo (Polshakov Dmitry) wrote…</i></summary><blockquote>

It's part of `getaddrinfo` [docs](https://linux.die.net/man/3/getaddrinfo). So it will be retranslated to `getaddrinfo` with their node meaning
</blockquote></details>

There should be a `,` after `host`. Also mention that only hostname lookup is done and the port is ignored.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/toktok/c-toxcore/461)*

**Comment by Diadlo** [comment]
Review status: 28 of 30 files reviewed at latest revision, 2 unresolved discussions.

---

*[toxcore/crypto_core.c, line 217 at r4](https://reviewable.io:443/reviews/toktok/c-toxcore/461#-Kc44xGR_h0FxkKvhxfQ:-KcDP39u8tM3_nqkHp6Y:b-896fix) ([raw file](https://github.com/toktok/c-toxcore/blob/a477e838a5a31c67d250460475a94bb31b57b596/toxcore/crypto_core.c#L217)):*
<details><summary><i>Previously, sudden6 wrote…</i></summary><blockquote>

sorry I forgot one line, `((num>>24)&0x000000FF) | // move byte 3 to byte 0` has to be the first one
</blockquote></details>

Done.

---

*[toxcore/crypto_core.c, line 219 at r6](https://reviewable.io:443/reviews/toktok/c-toxcore/461#-KcCbZNpNTUfKodhRYYw:-KcDP2rIbwRW7piiTq-y:b-896fix) ([raw file](https://github.com/toktok/c-toxcore/blob/2f8b53af89a6b25957ca647eb09446257d017ed9/toxcore/crypto_core.c#L219)):*
<details><summary><i>Previously, sudden6 wrote…</i></summary><blockquote>

maybe you could switch this line and the line above, so it's 3,2,1,0
</blockquote></details>

Done.

---

*[toxcore/network.h, line 388 at r3](https://reviewable.io:443/reviews/toktok/c-toxcore/461#-KbbD-rMmDRMrjVAKLQZ:-KcDP3ygpOyug0i0vBdx:b-896fix) ([raw file](https://github.com/toktok/c-toxcore/blob/30a7fbb692bb4bc80b936939d5776b79706226f7/toxcore/network.h#L388)):*
<details><summary><i>Previously, sudden6 wrote…</i></summary><blockquote>

There should be a `,` after `host`. Also mention that only hostname lookup is done and the port is ignored.
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/toktok/c-toxcore/461)*

**Comment by sudden6** [comment]
Reviewed 1 of 2 files at r7.
Review status: 29 of 30 files reviewed at latest revision, 1 unresolved discussion, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/toktok/c-toxcore/461)*

**Comment by sudden6** [comment]
:lgtm_strong:

---

Reviewed 1 of 1 files at r8.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/toktok/c-toxcore/461#-:-KcDSUxYLP7Lm4EUPijI:b-jbsf8z)*

**Comment by Diadlo** [comment]
@iphydf ping

**Comment by iphydf** [comment]
Ok, can you rebase and squash?

**Comment by Diadlo** [comment]
@iphydf Ok. Will do it a bit later today

**Comment by Diadlo** [comment]
@iphydf Done

**Comment by nurupo** [comment]
@Diadlo you didn't astyle your code, please fix https://travis-ci.org/TokTok/c-toxcore/jobs/205594073#L2043-L2247.

**Comment by nurupo** [comment]
*[other/bootstrap_daemon/src/tox-bootstrapd.c, line 315 at r9](https://reviewable.io:443/reviews/toktok/c-toxcore/461#-Ke0f06CqqR7HydDHSBh:-Ke0f06CqqR7HydDHSBi:bsswm18) ([raw file](https://github.com/toktok/c-toxcore/blob/1387c8f15032cab1af1c6444621b62af8d3a5494/other/bootstrap_daemon/src/tox-bootstrapd.c#L315)):*
> ```C
> 
>     uint64_t last_LANdiscovery = 0;
>     const uint16_t net_htons_port = net_htons(port);
> ```

The variable shouldn't have been renamed.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/toktok/c-toxcore/461)*
